tests/app/services/test_request_supplementary_data.py

```diff
@@ -1,20 +1,29 @@
 import pytest
 import responses
 from flask import Flask, current_app
+from marshmallow import ValidationError
 from requests import RequestException
+from sdc.crypto.key_store import KeyStore
 
 from app.services.supplementary_data import (
     SUPPLEMENTARY_DATA_REQUEST_MAX_RETRIES,
+    InvalidSupplementaryData,
+    MissingSupplementaryDataKey,
     SupplementaryDataRequestFailed,
+    decrypt_supplementary_data,
     get_supplementary_data,
 )
 from tests.app.utilities.test_schema import get_mocked_make_request
 
 TEST_SDS_URL = "http://test.domain/v1/unit_data"
+EXPECTED_SDS_DECRYPTION_VALIDATION_ERROR = (
+    "Supplementary data response cannot be decrypted without the key 'data'"
+)
 
-mock_supplementary_data_payload = {
+decrypted_mock_supplementary_data_payload = {
     "dataset_id": "44f1b432-9421-49e5-bd26-e63e18a30b69",
     "survey_id": "123",
+    "encryption_key_id": "df88fdad2612ae1e80571120e6c6371f55896696",
```

Comment:
based off above comments, maybe this key isn't needed? though I am unsure where it comes in instead of this

Reply:
Reason given in Mebin's comment above